Compiler error with typescript@4.8.4, sequelize@6.23.2, sequelize-typescript@2.1.3

Getting compiler error with latest versions of typescript, sequelize and sequelize-typescript:

`typescript`: `4.8.4`
`sequelize`: `6.23.2`
`sequelize-typescript`: `2.1.3`

```
node_modules/sequelize-typescript/dist/model/model/model.d.ts:7:131 - error TS2344: Type 'TModelAttributes' does not satisfy the constraint '{}'.

7 export declare type ModelType<TCreationAttributes, TModelAttributes> = new (values?: TCreationAttributes, options?: any) => Model<TModelAttributes, TCreationAttributes>;
                                                                                                                                    ~~~~~~~~~~~~~~~~

  node_modules/sequelize-typescript/dist/model/model/model.d.ts:7:52
    7 export declare type ModelType<TCreationAttributes, TModelAttributes> = new (values?: TCreationAttributes, options?: any) => Model<TModelAttributes, TCreationAttributes>;
                                                         ~~~~~~~~~~~~~~~~
    This type parameter might need an `extends {}` constraint.

Found 1 error in node_modules/sequelize-typescript/dist/model/model/model.d.ts:7
```

--- a/package.json
+++ b/package.json
@@ -137,7 +137,7 @@
     "source-map-support": "0.5.21",
     "sqlite3": "5.0.11",
     "ts-node": "10.9.1",
-    "typescript": "4.7.4",
+    "typescript": "4.8.4",
     "uuid-validate": "0.0.3"
   },
   "peerDependencies": {

--- a/src/associations/belongs-to-many/belongs-to-many-association.ts
+++ b/src/associations/belongs-to-many/belongs-to-many-association.ts
@@ -10,10 +10,10 @@ import { ModelType } from '../../model/model/model';
 import { ThroughOptions } from '../through/through-options';
 
 export class BelongsToManyAssociation<
-  TCreationAttributes,
-  TModelAttributes,
-  TCreationAttributesThrough,
-  TModelAttributesThrough
+  TCreationAttributes extends {},
+  TModelAttributes extends {},
+  TCreationAttributesThrough extends {},
+  TModelAttributesThrough extends {}
 > extends BaseAssociation<TCreationAttributes, TModelAttributes> {
   constructor(
     associatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>,

--- a/src/associations/belongs-to-many/belongs-to-many-options.ts
+++ b/src/associations/belongs-to-many/belongs-to-many-options.ts
@@ -2,7 +2,10 @@ import { BelongsToManyOptions as OriginBelongsToManyOptions } from 'sequelize';
 import { ModelClassGetter } from '../../model/shared/model-class-getter';
 import { ThroughOptions } from '../through/through-options';
 
-export type BelongsToManyOptions<TCreationAttributesThrough extends {}, TModelAttributesThrough extends {}> = {
+export type BelongsToManyOptions<
+  TCreationAttributesThrough extends {},
+  TModelAttributesThrough extends {}
+> = {
   [K in keyof OriginBelongsToManyOptions]: K extends 'through'
     ?
         | ModelClassGetter<TCreationAttributesThrough, TModelAttributesThrough>

--- a/src/associations/belongs-to-many/belongs-to-many-options.ts
+++ b/src/associations/belongs-to-many/belongs-to-many-options.ts
@@ -2,7 +2,7 @@ import { BelongsToManyOptions as OriginBelongsToManyOptions } from 'sequelize';
 import { ModelClassGetter } from '../../model/shared/model-class-getter';
 import { ThroughOptions } from '../through/through-options';
 
-export type BelongsToManyOptions<TCreationAttributesThrough, TModelAttributesThrough> = {
+export type BelongsToManyOptions<TCreationAttributesThrough extends {}, TModelAttributesThrough extends {}> = {
   [K in keyof OriginBelongsToManyOptions]: K extends 'through'
     ?
         | ModelClassGetter<TCreationAttributesThrough, TModelAttributesThrough>

--- a/src/associations/belongs-to-many/belongs-to-many.ts
+++ b/src/associations/belongs-to-many/belongs-to-many.ts
@@ -4,10 +4,10 @@ import { ModelClassGetter } from '../../model/shared/model-class-getter';
 import { addAssociation } from '../shared/association-service';
 
 export function BelongsToMany<
-  TCreationAttributes,
-  TModelAttributes,
-  TCreationAttributesThrough,
-  TModelAttributesThrough
+  TCreationAttributes extends {},
+  TModelAttributes extends {},
+  TCreationAttributesThrough extends {},
+  TModelAttributesThrough extends {}
 >(
   associatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>,
   through: ModelClassGetter<TCreationAttributesThrough, TModelAttributesThrough> | string,
@@ -15,19 +15,19 @@ export function BelongsToMany<
   otherKey?: string
 ): Function;
 export function BelongsToMany<
-  TCreationAttributes,
-  TModelAttributes,
-  TCreationAttributesThrough,
-  TModelAttributesThrough
+  TCreationAttributes extends {},
+  TModelAttributes extends {},
+  TCreationAttributesThrough extends {},
+  TModelAttributesThrough extends {}
 >(
   associatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>,
   options: BelongsToManyOptions<TCreationAttributesThrough, TModelAttributesThrough>
 ): Function;
 export function BelongsToMany<
-  TCreationAttributes,
-  TModelAttributes,
-  TCreationAttributesThrough,
-  TModelAttributesThrough
+  TCreationAttributes extends {},
+  TModelAttributes extends {},
+  TCreationAttributesThrough extends {},
+  TModelAttributesThrough extends {}
 >(
   associatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>,
   throughOrOptions:

--- a/src/associations/belongs-to/belongs-to-association.ts
+++ b/src/associations/belongs-to/belongs-to-association.ts
@@ -7,7 +7,7 @@ import { Association } from '../shared/association';
 import { ModelType } from '../../model/model/model';
 import { UnionAssociationOptions } from '../shared/union-association-options';
 
-export class BelongsToAssociation<TCreationAttributes, TModelAttributes> extends BaseAssociation<
+export class BelongsToAssociation<TCreationAttributes extends {}, TModelAttributes extends {}> extends BaseAssociation<
   TCreationAttributes,
   TModelAttributes
 > {

--- a/src/associations/belongs-to/belongs-to-association.ts
+++ b/src/associations/belongs-to/belongs-to-association.ts
@@ -7,10 +7,10 @@ import { Association } from '../shared/association';
 import { ModelType } from '../../model/model/model';
 import { UnionAssociationOptions } from '../shared/union-association-options';
 
-export class BelongsToAssociation<TCreationAttributes extends {}, TModelAttributes extends {}> extends BaseAssociation<
-  TCreationAttributes,
-  TModelAttributes
-> {
+export class BelongsToAssociation<
+  TCreationAttributes extends {},
+  TModelAttributes extends {}
+> extends BaseAssociation<TCreationAttributes, TModelAttributes> {
   constructor(
     associatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>,
     protected options: BelongsToOptions

--- a/src/associations/belongs-to/belongs-to.ts
+++ b/src/associations/belongs-to/belongs-to.ts
@@ -4,17 +4,17 @@ import { BelongsToAssociation } from './belongs-to-association';
 import { ModelClassGetter } from '../../model/shared/model-class-getter';
 import { addAssociation, getPreparedAssociationOptions } from '../shared/association-service';
 
-export function BelongsTo<TCreationAttributes, TModelAttributes>(
+export function BelongsTo<TCreationAttributes extends {}, TModelAttributes extends {}>(
   associatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>,
   foreignKey?: string
 ): Function;
 
-export function BelongsTo<TCreationAttributes, TModelAttributes>(
+export function BelongsTo<TCreationAttributes extends {}, TModelAttributes extends {}>(
   associatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>,
   options?: BelongsToOptions
 ): Function;
 
-export function BelongsTo<TCreationAttributes, TModelAttributes>(
+export function BelongsTo<TCreationAttributes extends {}, TModelAttributes extends {}>(
   associatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>,
   optionsOrForeignKey?: string | BelongsToOptions
 ): Function {

--- a/src/associations/foreign-key/foreign-key-meta.ts
+++ b/src/associations/foreign-key/foreign-key-meta.ts
@@ -1,6 +1,6 @@
 import { ModelClassGetter } from '../../model/shared/model-class-getter';
 
-export interface ForeignKeyMeta<TCreationAttributes, TModelAttributes> {
+export interface ForeignKeyMeta<TCreationAttributes extends {}, TModelAttributes extends {}> {
   relatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>;
   foreignKey: string;
 }

--- a/src/associations/foreign-key/foreign-key-service.ts
+++ b/src/associations/foreign-key/foreign-key-service.ts
@@ -7,10 +7,10 @@ import { ModelType } from '../../model/model/model';
 const FOREIGN_KEYS_KEY = 'sequelize:foreignKeys';
 
 export function getForeignKeyOptions<
-  TCreationAttributes,
-  TModelAttributes,
-  TCreationAttributesThrough,
-  TModelAttributesThrough
+  TCreationAttributes extends {},
+  TModelAttributes extends {},
+  TCreationAttributesThrough extends {},
+  TModelAttributesThrough extends {}
 >(
   relatedClass: ModelType<TCreationAttributes, TModelAttributes>,
   classWithForeignKey?: ModelType<TCreationAttributesThrough, TModelAttributesThrough>,
@@ -48,7 +48,7 @@ export function getForeignKeyOptions<
 /**
  * Adds foreign key meta data for specified class
  */
-export function addForeignKey<TCreationAttributes, TModelAttributes>(
+export function addForeignKey<TCreationAttributes extends {}, TModelAttributes extends {}>(
   target: any,
   relatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>,
   foreignKey: string
@@ -67,7 +67,7 @@ export function addForeignKey<TCreationAttributes, TModelAttributes>(
 /**
  * Returns foreign key meta data from specified class
  */
-export function getForeignKeys<TCreationAttributes, TModelAttributes>(
+export function getForeignKeys<TCreationAttributes extends {}, TModelAttributes extends {}>(
   target: any
 ): ForeignKeyMeta<TCreationAttributes, TModelAttributes>[] | undefined {
   const foreignKeys = Reflect.getMetadata(FOREIGN_KEYS_KEY, target);

--- a/src/associations/foreign-key/foreign-key.ts
+++ b/src/associations/foreign-key/foreign-key.ts
@@ -1,7 +1,7 @@
 import { addForeignKey } from './foreign-key-service';
 import { ModelClassGetter } from '../../model/shared/model-class-getter';
 
-export function ForeignKey<TCreationAttributes, TModelAttributes>(
+export function ForeignKey<TCreationAttributes extends {}, TModelAttributes extends {}>(
   relatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>
 ): Function {
   return (target: any, propertyName: string) => {

--- a/src/associations/has/has-association.ts
+++ b/src/associations/has/has-association.ts
@@ -7,7 +7,7 @@ import { Association } from '../shared/association';
 import { UnionAssociationOptions } from '../shared/union-association-options';
 import { ModelType } from '../../model/model/model';
 
-export class HasAssociation<TCreationAttributes, TModelAttributes> extends BaseAssociation<
+export class HasAssociation<TCreationAttributes extends {}, TModelAttributes extends {}> extends BaseAssociation<
   TCreationAttributes,
   TModelAttributes
 > {

--- a/src/associations/has/has-association.ts
+++ b/src/associations/has/has-association.ts
@@ -7,10 +7,10 @@ import { Association } from '../shared/association';
 import { UnionAssociationOptions } from '../shared/union-association-options';
 import { ModelType } from '../../model/model/model';
 
-export class HasAssociation<TCreationAttributes extends {}, TModelAttributes extends {}> extends BaseAssociation<
-  TCreationAttributes,
-  TModelAttributes
-> {
+export class HasAssociation<
+  TCreationAttributes extends {},
+  TModelAttributes extends {}
+> extends BaseAssociation<TCreationAttributes, TModelAttributes> {
   constructor(
     associatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>,
     protected options: HasManyOptions | HasOneOptions,

--- a/src/associations/has/has-many.ts
+++ b/src/associations/has/has-many.ts
@@ -5,17 +5,17 @@ import { ModelClassGetter } from '../../model/shared/model-class-getter';
 import { addAssociation, getPreparedAssociationOptions } from '../shared/association-service';
 import { Association } from '../shared/association';
 
-export function HasMany<TCreationAttributes, TModelAttributes>(
+export function HasMany<TCreationAttributes extends {}, TModelAttributes extends {}>(
   associatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>,
   foreignKey?: string
 ): Function;
 
-export function HasMany<TCreationAttributes, TModelAttributes>(
+export function HasMany<TCreationAttributes extends {}, TModelAttributes extends {}>(
   associatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>,
   options?: HasManyOptions
 ): Function;
 
-export function HasMany<TCreationAttributes, TModelAttributes>(
+export function HasMany<TCreationAttributes extends {}, TModelAttributes extends {}>(
   associatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>,
   optionsOrForeignKey?: string | HasManyOptions
 ): Function {

--- a/src/associations/has/has-one.ts
+++ b/src/associations/has/has-one.ts
@@ -10,7 +10,7 @@ export function HasOne<TCreationAttributes extends {}, TModelAttributes extends 
   foreignKey?: string
 ): Function;
 
-export function HasOne<TCreationAttributes extends {}, TModelAttributes  extends {}>(
+export function HasOne<TCreationAttributes extends {}, TModelAttributes extends {}>(
   associatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>,
   options?: HasOneOptions
 ): Function;

--- a/src/associations/has/has-one.ts
+++ b/src/associations/has/has-one.ts
@@ -5,17 +5,17 @@ import { ModelClassGetter } from '../../model/shared/model-class-getter';
 import { addAssociation, getPreparedAssociationOptions } from '../shared/association-service';
 import { Association } from '../shared/association';
 
-export function HasOne<TCreationAttributes, TModelAttributes>(
+export function HasOne<TCreationAttributes extends {}, TModelAttributes extends {}>(
   associatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>,
   foreignKey?: string
 ): Function;
 
-export function HasOne<TCreationAttributes, TModelAttributes>(
+export function HasOne<TCreationAttributes extends {}, TModelAttributes  extends {}>(
   associatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>,
   options?: HasOneOptions
 ): Function;
 
-export function HasOne<TCreationAttributes, TModelAttributes>(
+export function HasOne<TCreationAttributes extends {}, TModelAttributes extends {}>(
   associatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>,
   optionsOrForeignKey?: string | HasOneOptions
 ): Function {

--- a/src/associations/shared/association-service.ts
+++ b/src/associations/shared/association-service.ts
@@ -28,7 +28,7 @@ export function getPreparedAssociationOptions(
 /**
  * Stores association meta data for specified class
  */
-export function addAssociation<TCreationAttributes, TModelAttributes>(
+export function addAssociation<TCreationAttributes extends {}, TModelAttributes extends {}>(
   target: any,
   association: BaseAssociation<TCreationAttributes, TModelAttributes>
 ): void {
@@ -44,7 +44,7 @@ export function addAssociation<TCreationAttributes, TModelAttributes>(
 /**
  * Returns association meta data from specified class
  */
-export function getAssociations<TCreationAttributes, TModelAttributes>(
+export function getAssociations<TCreationAttributes extends {}, TModelAttributes extends {}>(
   target: any
 ): BaseAssociation<TCreationAttributes, TModelAttributes>[] | undefined {
   const associations = Reflect.getMetadata(ASSOCIATIONS_KEY, target);
@@ -53,14 +53,14 @@ export function getAssociations<TCreationAttributes, TModelAttributes>(
   }
 }
 
-export function setAssociations<TCreationAttributes, TModelAttributes>(
+export function setAssociations<TCreationAttributes extends {}, TModelAttributes extends {}>(
   target: any,
   associations: BaseAssociation<TCreationAttributes, TModelAttributes>[]
 ): void {
   Reflect.defineMetadata(ASSOCIATIONS_KEY, associations, target);
 }
 
-export function getAssociationsByRelation<TCreationAttributes, TModelAttributes>(
+export function getAssociationsByRelation<TCreationAttributes extends {}, TModelAttributes extends {}>(
   target: any,
   relatedClass: any
 ): BaseAssociation<TCreationAttributes, TModelAttributes>[] {

--- a/src/associations/shared/association-service.ts
+++ b/src/associations/shared/association-service.ts
@@ -60,10 +60,10 @@ export function setAssociations<TCreationAttributes extends {}, TModelAttributes
   Reflect.defineMetadata(ASSOCIATIONS_KEY, associations, target);
 }
 
-export function getAssociationsByRelation<TCreationAttributes extends {}, TModelAttributes extends {}>(
-  target: any,
-  relatedClass: any
-): BaseAssociation<TCreationAttributes, TModelAttributes>[] {
+export function getAssociationsByRelation<
+  TCreationAttributes extends {},
+  TModelAttributes extends {}
+>(target: any, relatedClass: any): BaseAssociation<TCreationAttributes, TModelAttributes>[] {
   const associations = getAssociations<TCreationAttributes, TModelAttributes>(target);
   return (associations || []).filter((association) => {
     const _relatedClass = association.getAssociatedClass();

--- a/src/associations/shared/base-association.ts
+++ b/src/associations/shared/base-association.ts
@@ -4,7 +4,7 @@ import { ModelClassGetter } from '../../model/shared/model-class-getter';
 import { ModelType } from '../../model/model/model';
 import { Sequelize } from '../../sequelize/sequelize/sequelize';
 
-export abstract class BaseAssociation<TCreationAttributes, TModelAttributes> {
+export abstract class BaseAssociation<TCreationAttributes extends {}, TModelAttributes extends {}> {
   constructor(
     private associatedClassGetter: ModelClassGetter<TCreationAttributes, TModelAttributes>,
     protected options: UnionAssociationOptions

--- a/src/associations/through/through-options.ts
+++ b/src/associations/through/through-options.ts
@@ -1,7 +1,7 @@
 import { ThroughOptions as OriginThroughOptions } from 'sequelize';
 import { ModelClassGetter } from '../../model/shared/model-class-getter';
 
-export type ThroughOptions<TCreationAttributes, TModelAttributes> = {
+export type ThroughOptions<TCreationAttributes extends {}, TModelAttributes extends {}> = {
   [K in keyof OriginThroughOptions]: K extends 'model'
     ? ModelClassGetter<TCreationAttributes, TModelAttributes> | string
     : OriginThroughOptions[K];

--- a/src/model/model/model.ts
+++ b/src/model/model/model.ts
@@ -16,7 +16,7 @@ import { AssociationActionOptions } from './association/association-action-optio
 import { AssociationCreateOptions } from './association/association-create-options';
 import { Repository } from '../../sequelize/repository/repository';
 
-export type ModelType<TCreationAttributes, TModelAttributes> = new (
+export type ModelType<TCreationAttributes extends {}, TModelAttributes extends {}> = new (
   values?: TCreationAttributes,
   options?: any
 ) => Model<TModelAttributes, TCreationAttributes>;

--- a/src/model/shared/model-class-getter.ts
+++ b/src/model/shared/model-class-getter.ts
@@ -1,5 +1,5 @@
 import { ModelType } from '../model/model';
 
-export type ModelClassGetter<TCreationAttributes, TModelAttributes> = (
+export type ModelClassGetter<TCreationAttributes extends {}, TModelAttributes extends {}> = (
   returns?: void
 ) => ModelType<TCreationAttributes, TModelAttributes>;

--- a/src/scopes/default-scope.ts
+++ b/src/scopes/default-scope.ts
@@ -11,14 +11,14 @@ export function DefaultScope(scopeGetter: DefaultScopeGetter): Function;
  * Decorator for defining default Model scope
  * @deprecated
  */
-export function DefaultScope<TCreationAttributes, TModelAttributes>(
+export function DefaultScope<TCreationAttributes extends {}, TModelAttributes extends {}>(
   scope: ScopeFindOptions<TCreationAttributes, TModelAttributes>
 ): Function;
 
 /**
  * Decorator for defining default Model scope
  */
-export function DefaultScope<TCreationAttributes, TModelAttributes>(
+export function DefaultScope<TCreationAttributes extends {}, TModelAttributes extends {}>(
   scopeOrSsopeGetter: ScopeFindOptions<TCreationAttributes, TModelAttributes> | DefaultScopeGetter
 ): Function {
   return (target: any) => {

--- a/src/scopes/scope-find-options.ts
+++ b/src/scopes/scope-find-options.ts
@@ -1,7 +1,7 @@
 import { Association, FindOptions, IncludeOptions } from 'sequelize';
 import { ModelClassGetter } from '../model/shared/model-class-getter';
 
-export type ScopeIncludeOptions<TCreationAttributes, TModelAttributes> = {
+export type ScopeIncludeOptions<TCreationAttributes extends {}, TModelAttributes extends {}> = {
   [K in keyof IncludeOptions]: K extends 'model'
     ? ModelClassGetter<TCreationAttributes, TModelAttributes>
     : K extends 'include'
@@ -9,7 +9,7 @@ export type ScopeIncludeOptions<TCreationAttributes, TModelAttributes> = {
     : IncludeOptions[K];
 };
 
-export type ScopeFindOptions<TCreationAttributes, TModelAttributes> = {
+export type ScopeFindOptions<TCreationAttributes extends {}, TModelAttributes extends {}> = {
   [K in keyof FindOptions]: K extends 'include'
     ?
         | ModelClassGetter<TCreationAttributes, TModelAttributes>[]

--- a/src/scopes/scope-options.ts
+++ b/src/scopes/scope-options.ts
@@ -2,7 +2,7 @@ import { ScopeTableOptions } from './scope-table-options';
 import { ScopeFindOptions } from './scope-find-options';
 import { FindOptions } from 'sequelize';
 
-export interface ScopeOptions<TCreationAttributes, TModelAttributes>
+export interface ScopeOptions<TCreationAttributes extends {}, TModelAttributes extends {}>
   extends ScopeTableOptions<TCreationAttributes, TModelAttributes> {
   defaultScope?: ScopeFindOptions<TCreationAttributes, TModelAttributes>;
 }

--- a/src/scopes/scope-service.ts
+++ b/src/scopes/scope-service.ts
@@ -66,7 +66,7 @@ export const resolvesDeprecatedScopes = (model: ModelCtor): void => {
  * Adds scope option meta data for specified prototype
  * @deprecated
  */
-export function addScopeOptions<TCreationAttributes, TModelAttributes>(
+export function addScopeOptions<TCreationAttributes extends {}, TModelAttributes extends {}>(
   target: any,
   options: ScopeOptions<TCreationAttributes, TModelAttributes>
 ): void {
@@ -78,7 +78,7 @@ export function addScopeOptions<TCreationAttributes, TModelAttributes>(
  * Returns scope option meta data from specified target
  * @deprecated
  */
-export function getScopeOptions<TCreationAttributes, TModelAttributes>(
+export function getScopeOptions<TCreationAttributes extends {}, TModelAttributes extends {}>(
   target: any
 ): ScopeOptions<TCreationAttributes, TModelAttributes> | undefined {
   const options = Reflect.getMetadata(SCOPES_KEY, target);
@@ -90,7 +90,7 @@ export function getScopeOptions<TCreationAttributes, TModelAttributes>(
 /**
  * @deprecated
  */
-function resolveDeprecatedScope<TCreationAttributes, TModelAttributes>(
+function resolveDeprecatedScope<TCreationAttributes extends {}, TModelAttributes extends {}>(
   scopeName: string,
   model: ModelCtor,
   options: ScopeFindOptions<TCreationAttributes, TModelAttributes> | Function | undefined
@@ -108,7 +108,7 @@ function resolveDeprecatedScope<TCreationAttributes, TModelAttributes>(
  * Set scope option meta data for specified prototype
  * @deprecated
  */
-function setScopeOptions<TCreationAttributes, TModelAttributes>(
+function setScopeOptions<TCreationAttributes extends {}, TModelAttributes extends {}>(
   target: any,
   options: ScopeOptions<TCreationAttributes, TModelAttributes>
 ): void {

--- a/src/scopes/scope-table-options.ts
+++ b/src/scopes/scope-table-options.ts
@@ -1,6 +1,6 @@
 import { ScopeFindOptions } from './scope-find-options';
 
-export interface ScopeTableOptions<TCreationAttributes, TModelAttributes> {
+export interface ScopeTableOptions<TCreationAttributes extends {}, TModelAttributes extends {}> {
   [scopeName: string]:
     | ScopeFindOptions<TCreationAttributes, TModelAttributes>
     | Function

--- a/src/scopes/scopes.ts
+++ b/src/scopes/scopes.ts
@@ -11,14 +11,14 @@ export function Scopes(scopesGetter: ScopesOptionsGetter): Function;
  * Decorator for defining Model scopes
  * @deprecated
  */
-export function Scopes<TCreationAttributes, TModelAttributes>(
+export function Scopes<TCreationAttributes extends {}, TModelAttributes extends {}>(
   scopes: ScopeTableOptions<TCreationAttributes, TModelAttributes>
 ): Function;
 
 /**
  * Decorator for defining Model scopes
  */
-export function Scopes<TCreationAttributes, TModelAttributes>(
+export function Scopes<TCreationAttributes extends {}, TModelAttributes extends {}>(
   scopesOrScopesGetter:
     | ScopeTableOptions<TCreationAttributes, TModelAttributes>
     | ScopesOptionsGetter

--- a/src/sequelize/sequelize/sequelize.ts
+++ b/src/sequelize/sequelize/sequelize.ts
@@ -32,7 +32,7 @@ export class Sequelize extends OriginSequelize {
     }
   }
 
-  model<TCreationAttributes, TModelAttributes>(
+  model<TCreationAttributes extends {}, TModelAttributes extends {}>(
     model: string | ModelType<TCreationAttributes, TModelAttributes>
   ): ModelCtor {
     if (typeof model !== 'string') {

